Sources/Fuzzilli/Lifting/JavaScriptLifter.swift

```diff
@@ -115,6 +115,44 @@ public class JavaScriptLifter: Lifter {
                 return expr(for: instr.input(idx))
             }
 
+            // Helper function to lift destruct array operations
+            func liftArrayPattern(operation: Operation) -> String {
+                var arrayPattern = ""
+                var outputs: [String] = []
+                var outputIndex = 0
+                var indices: [Int] = []
+                var hasRestElement: Bool = false
+                switch operation {
+                    case let op as DestructArray:
+                        outputs = instr.outputs.map({ $0.identifier })
+                        indices = op.indices
+                        hasRestElement = op.hasRestElement
+                    case let op as DestructArrayAndReassign:
+                        outputs = instr.inputs.dropFirst().map({ $0.identifier })
+                        indices = op.indices
+                        hasRestElement = op.hasRestElement
+                    default:
+                        return arrayPattern
+                }
+                assert(indices.count == outputs.count)
+                if indices.count > 0 {
+                    for index in 0...indices.last! {
+                        if indices.contains(index) {
+                            if index == indices.last! {
+                                hasRestElement ? arrayPattern.append(" ...\(outputs[outputIndex])") : arrayPattern.append(" \(outputs[outputIndex])")
+                            } else {
+                                arrayPattern.append("\(outputs[Int(outputIndex)]),")
+                            }
+                            outputIndex += 1
+                        } else {
+                            arrayPattern.append(" ,")
+                        }
+                    }
+                }
```

Comment:
Can we simplify this a bit and make it O(N) (not O(N^2), not that it really matters though). Something like
```swift
var lastIndex = 0
for (index, output) in zip(indices, outputs) {
    let skipped = index - lastIndex
    lastIndex = index
    let dots = index == indices.last! && hasRestElement ? "..." : ""
    arrayPattern += String(repeating: ",", count: skipped) + dots + output + ","
}
```
maybe?

Reply:
Nice! I learn a new swift construct with each PR I send :D
I'll update this logic.